### PR TITLE
Update HOMME Aurora machine cmake config

### PR DESCRIPTION
**Pull request by abagusetty**
0. Update Cmake machine config for Aurora
1. Cleanup certain warnings related to headers and Intel compilers
2. Linking Fortran/C++/SYCL with CXX linker (-fortlib) instead of Fortran language for SYCL builds. Since Fortran linking with SYCL is no longer supported (https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2024-2/fortlib.html)

### CODE REVIEW

File: components/homme/src/share/cxx/ExecSpaceDefs.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #endif
 
 #ifdef KOKKOS_ENABLE_SYCL
-#include <CL/sycl.hpp>
+#include <sycl/sycl.hpp>
```
Comment (oksanaguba) [inline components/homme/src/share/cxx/ExecSpaceDefs.cpp:25]: could you explain this change please?

**Review by oksanaguba — APPROVED** [review]

File: components/homme/cmake/machineFiles/aurora-aot.cmake [hunk 2]
```diff
@@ -57,8 +50,6 @@ set (ENABLE_HORIZ_OPENMP OFF CACHE BOOL "")
 
 set (HOMME_TESTING_PROFILE "dev" CACHE STRING "")
 
-set (USE_NUM_PROCS 4 CACHE STRING "")
-
-SET (USE_MPI_OPTIONS "--bind-to core" CACHE FILEPATH "")
-
+set (USE_NUM_PROCS 12 CACHE STRING "")
 
+SET (USE_MPI_OPTIONS "--pmi=pmix --cpu-bind list:0-7,104-111:8-15,112-119:16-23,120-127:24-31,128-135:32-39,136-143:40-47,144-151:52-59,156-163:60-67,164-171:68-75,172-179:76-83,180-187:84-91,188-195:92-99,196-203 gpu_tile_compact.sh" CACHE FILEPATH "")
```
Comment (oksanaguba) [inline components/homme/cmake/machineFiles/aurora-aot.cmake:55]: is tile script always in a user's path?

File: components/homme/src/share/cxx/ExecSpaceDefs.cpp (hunk 1 above)
Comment (abagusetty) [reply components/homme/src/share/cxx/ExecSpaceDefs.cpp:25]: SYCL specification has long deprecated these headers. <CL/sycl.hpp> header was valid only in SYCL 1.2.1 but it has been deprecated for <sycl/sycl.hpp> in SYCL 2020 Standards. 

From the SYCL spec
```
For consistency, the programming API will only refer to the <sycl/sycl.hpp> header and the ::sycl namespace, but this should be considered synonymous with the SYCL 1.2.1 header and namespace.```

File: components/homme/cmake/machineFiles/aurora-aot.cmake (hunk 2 above)
Comment (abagusetty) [reply components/homme/cmake/machineFiles/aurora-aot.cmake:55]: Yes, the tile binding script is always in the SYSTEM path for the users

**Comment by rljacob** [comment]
To clarify, the changes in components/homme/cmake only impact the standalone build of HOMMEXX but the other changes could impact the EAMxx/SCREAM build?

**Comment by abagusetty** [comment]
> To clarify, the changes in components/homme/cmake only impact the standalone build of HOMMEXX but the other changes could impact the EAMxx/SCREAM build?

I guess @oksanaguba could fill-in more accurately, but changes outside the standalone cmake config to HOMME were to address deprecation warnings and so are cosmetic

**Comment by oksanaguba** [comment]
yes, the changes besides cmake will be picked up by scream builds, but as Abhishek says they are an improvement. SCREAM flags may need revision in a similar way (or not).

File: components/homme/src/share/cxx/vector/vector_pragmas.hpp [hunk 3]
```diff
@@ -7,7 +7,7 @@
 #ifndef HOMMEXX_VECTOR_PRAGMAS_HPP
 #define HOMMEXX_VECTOR_PRAGMAS_HPP
 
-#if defined(__INTEL_COMPILER)
+#if defined(__INTEL_COMPILER) || defined(__INTEL_CLANG_COMPILER) || defined(__INTEL_LLVM_COMPILER)
```
[the remark this answers is not in the document]
Comment (abagusetty) [reply components/homme/src/share/cxx/vector/vector_pragmas.hpp:10]: Intel compilers are identified as both clang and llvm based. Both the macros are defined by oneAPI compilers. 

The difference being the naming convention: If it is a compiler released by Intel as a product (both the macros are defined) or if a user builds the compiler themselves from upstream repo (then either or both could be defined)

**Comment by abagusetty** [comment]
@oksanaguba The CI seems to have failed. Is that what is blocking at the moment

**Review by oksanaguba — APPROVED** [review]

**Comment by oksanaguba** [comment]
@abagusetty we use a manual process to merge to "next" first and let cdash tests pass overnight https://my.cdash.org/index.php?project=E3SM . then the PR can go to "master". i will merge this to next maybe today. i am not sure about CI. thanks.

**Comment by rljacob** [comment]
@oksanaguba the CI we run on PRs through github actions had a fail which was not related to the PR.  I reran it and its fine.

**Comment by oksanaguba** [comment]
On chrysalis, homme build fails with
 5) intel-mkl/2020.4.304-g2qaxzf   
 4) intel/20.0.4-kodw73g      
because of lapack calls `undefined reference to `dgemm_'` etc. @abagusetty would  you want to revert mkl flag, and i will retest?

**Comment by abagusetty** [comment]
> On chrysalis, homme build fails with 5) intel-mkl/2020.4.304-g2qaxzf 4) intel/20.0.4-kodw73g because of lapack calls `undefined reference to `dgemm_'` etc. @abagusetty would you want to revert mkl flag, and i will retest?

Addressed in the following commit

**Comment by oksanaguba** [comment]
ran homme suites and one ERS test on chrysalis, all ok.

**Pull request merged**